### PR TITLE
feat: appointment Letter

### PR DESCRIPTION
**Pull request by Anurag810**
**Appointment Letter Template:**

image

**Appointment Letter: **

image
image
image

**Print Format:**
screencapture-human-resource-local-8011-desk-2019-12-27-11_48_50

### CODE REVIEW

**Comment by Anurag810** [comment]
@nabinhait please suggest changes.

File: erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+ "actions": [],
+ "autoname": "APP-LETTER-TEMP-.#####",
+ "creation": "2019-12-26 12:20:14.219578",
+ "doctype": "DocType",
+ "editable_grid": 1,
+ "engine": "InnoDB",
+ "field_order": [
+  "introduction",
+  "content"
```
Comment (nabinhait) [inline erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json]: Why not closing_note?

File: erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ "permissions": [
+  {
+   "create": 1,
+   "delete": 1,
+   "email": 1,
+   "export": 1,
+   "print": 1,
+   "read": 1,
+   "report": 1,
+   "role": "System Manager",
```
Comment (nabinhait) [inline erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json:49]: It should be HR Manager

File: erpnext/hr/doctype/appointment_letter/appointment_letter.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import frappe
+from frappe.model.document import Document
+from pprint import pprint
+
+class AppointmentLetter(Document):
+	pass
+
+@frappe.whitelist()
+def get_appointment_letter_details(template):
+	print(template)
```
Comment (nabinhait) [inline erpnext/hr/doctype/appointment_letter/appointment_letter.py]: remove print

File: erpnext/hr/doctype/appointment_letter/appointment_letter.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	body = []
+	intro= frappe.get_list("Appointment Letter Template",
+		fields = ['introduction'],
+		filters={'name': template
+	})[0]
+	content = frappe.get_list("Appointment Letter content",
+	fields = ['title', 'description'],
```
Comment (nabinhait) [inline erpnext/hr/doctype/appointment_letter/appointment_letter.py]: indent?

File: erpnext/hr/doctype/appointment_letter/appointment_letter.json [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+ "permissions": [
+  {
+   "create": 1,
+   "delete": 1,
+   "email": 1,
+   "export": 1,
+   "print": 1,
+   "read": 1,
+   "report": 1,
+   "role": "System Manager",
```
Comment (nabinhait) [inline erpnext/hr/doctype/appointment_letter/appointment_letter.json:104]: Should be HR User and HR Manager

**Review by nabinhait — CHANGES_REQUESTED** [review]
- Add Docs

**Review by nabinhait — CHANGES_REQUESTED** [review]
- Add link in HR home page

File: erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json (hunk 1 above)
Comment (Anurag810) [reply erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json]: Done

File: erpnext/hr/doctype/appointment_letter/appointment_letter.py (hunk 3 above)
Comment (Anurag810) [reply erpnext/hr/doctype/appointment_letter/appointment_letter.py]: Done

File: erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json (hunk 2 above)
Comment (Anurag810) [reply erpnext/hr/doctype/appointment_letter_template/appointment_letter_template.json:49]: Done

File: erpnext/hr/doctype/appointment_letter/appointment_letter.json (hunk 5 above)
Comment (Anurag810) [reply erpnext/hr/doctype/appointment_letter/appointment_letter.json:104]: Done

**Comment by Anurag810** [comment]
> * Add link in HR home page

done

**Comment by Anurag810** [comment]
docs  https://github.com/frappe/erpnext_com/pull/655

**Review by nabinhait — APPROVED** [review]

**Pull request merged**